cli/src/lib/website/read.ts

```diff
@@ -1,23 +1,40 @@
-import {
-  Args,
-  ArrayTypes,
-  SmartContract,
-  U32,
-  Web3Provider,
-} from '@massalabs/massa-web3'
+import { SmartContract, U32, Web3Provider } from '@massalabs/massa-web3'
 import { sha256 } from 'js-sha256'
-import { getChunkKey } from './chunk'
+import { TextDecoder } from 'util'
+
+import {
+  FILE_LOCATION_TAG,
+  fileChunkCountKey,
+  fileChunkKey,
+} from './storageKeys'
 
 /**
  * Lists files from the given website on Massa blockchain
  * @param sc - SmartContract instance
  * @returns List of file paths in the website
  */
-export async function listFiles(sc: SmartContract): Promise<string[]> {
-  const fileListRaw = await sc.read('getFilePathList', undefined)
-  const fileListArgs = new Args(fileListRaw.value)
+export async function listFiles(
+  provider: Web3Provider,
+  sc: SmartContract
+): Promise<string[]> {
+  const allStorageKeys = await provider.client.getDataStoreKeys(sc.address)
```

Comment:
You have a function provider.getStorageKeys if you update the version next of massa-web3